src/crossover.rs

```diff
@@ -55,4 +62,60 @@ impl Crossover {
     pub fn encrypted_data(&self) -> &PoseidonCipher {
         &self.encrypted_data
     }
+
+    /// Converts a Crossover into it's byte representation
+    pub fn to_bytes(&self) -> [u8; Crossover::serialized_size()] {
+        let mut buf = [0u8; Crossover::serialized_size()];
+        let mut n = 0;
+
+        buf[n..n + 32].copy_from_slice(
+            &JubJubAffine::from(&self.value_commitment).to_bytes()[..],
+        );
+        n += 32;
+
+        buf[n..n + 32].copy_from_slice(&self.nonce.to_bytes()[..]);
+        n += 32;
+
+        buf[n..n + PoseidonCipher::cipher_size_bytes()]
+            .copy_from_slice(&self.encrypted_data.to_bytes()[..]);
+        n += PoseidonCipher::cipher_size_bytes();
+
+        assert_eq!(n, Crossover::serialized_size());
```

Comment:
We can't have assertions in the middle of the code.

There is a better option `debug_assert` that will run these assertions in debug mode(non-optimized compilation)

Reply:
It depends, sometimes is actually helps the compiler to optimize code; but if it's used at the beginning, not in this case of course. Agreed it should be done a `debug_assert` or skipped at all.

Reply:
I thought the same while coding it. But the assertions were already in other fn's. So I basically went on the same direction. Will use `debug_assert!` as I did with zerocaf back then.